mobile/lib/shared/services/immich_logger.service.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:io';
+
+import 'package:flutter/widgets.dart';
+import 'package:hive/hive.dart';
+import 'package:immich_mobile/constants/hive_box.dart';
+import 'package:immich_mobile/shared/models/in_app_logger_message.model.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:share_plus/share_plus.dart';
+
+class ImmichLogger {
+  final String logContext;
+  final maxLogEntries = 200;
+
+  final Box<InAppLoggerMessage> _box = Hive.box(inAppLoggerBox);
+
+  List<InAppLoggerMessage> get messages =>
+      _box.values.toList().reversed.toList();
+
+  ImmichLogger(this.logContext) {
+    if (_box.length > maxLogEntries) {
+      var indexesToBeDeleted = _box.length - maxLogEntries;
+      for (var i = 0; i < indexesToBeDeleted; i++) {
+        _box.deleteAt(indexesToBeDeleted);
+      }
+    }
+  }
+
+  void log(message, {String type = "info", String? additionalContext}) {
```

Comment:
`type` should be an `Enum` called e.g. `Level`

Reply:
Additionally, log methods named after their level are nice-to-have (but not required). Eg (pseudocodey):
```
void error(message) {
  log(message, type = "error");
}
```

Reply:
using the `logging`package would get us these  methods named according to their level